Modules/fact.py

```diff
@@ -0,0 +1,225 @@
+"""
+fact.py - Fact Class
+
+Provides a Fact object for the Fact Manager Class.
+
+Copyright (c) 2018 The Fuel Rat Mischief,
+All rights reserved.
+
+Licensed under the BSD 3-Clause License.
+
+See LICENSE.md
+"""
+
+import datetime
+
+
+class Fact:
+    def __init__(self, name=None, lang='en', message=None,
+                 aliases=None, author=None, edited=None,
+                 editedby=None, mfd=False):
+        self._name = name
+        self._lang = lang
+        self._message = message
+        self._aliases = aliases
+        self._author = author
+        self._editedby = editedby
+        self._edited = edited if edited else datetime.datetime.now(datetime.timezone.utc)
+        self._mfd = mfd
+
+    @property
+    def name(self) -> str:
+        """
+        Name of fact, ie 'prep'
+        Returns: name of fact
+        """
+        return self._name
+
+    @property
+    def lang(self) -> str:
+        """
+        language code, ie 'en'
+        Returns: language of fact.
+
+        This may be may be longer than two, in the case of template facts.
+        """
+        return self._lang
+
+    @property
+    def message(self) -> str:
+        """
+        Message for fact - the set content.
+        Returns: str fact content
+        """
+        return self._message
+
+    @property
+    def aliases(self) -> list:
+        """
+        Registered Aliases for a command.  They are globally unique.
+        Returns: list of aliases for fact
+        """
+        return self._aliases
+
+    @property
+    def author(self) -> str:
+        """
+        Author of fact.
+        Returns: str author of fact.
+        """
+        return self._author
+
+    @property
+    def editedby(self) -> str:
+        """
+        Last editor of fact. Editor is the the rat who added the fact, if new.
+        Returns: str editor of fact.
+        """
+        return self._editedby
+
+    @property
+    def edited(self) -> datetime:
+        """
+        Datetime stamp for last edit of fact
+        Returns: datetime object
+        """
+        return self._edited
+
+    @property
+    def mfd(self) -> bool:
+        """
+        If Fact is marked for deletion, True/False
+        Returns: bool marked for deletion
+        """
+        return self._mfd
+
+    @name.setter
+    def name(self, value: str):
+        """
+        Sets fact name.
+        Args:
+            value: str
+        Returns: Nothing.
+        """
+        if not isinstance(value, str):
+            raise TypeError("Fact.name must be of string type.")
+
+        self._name = value
+
+    @lang.setter
+    def lang(self, value: str):
+        """
+        Sets fact language ID.
+        Args:
+            value: str language ID
+        Returns: Nothing
```

Comment:
newlines between blocks. Helpful hint, place the carrot over the function and hit `ctrl`+`q` 😉